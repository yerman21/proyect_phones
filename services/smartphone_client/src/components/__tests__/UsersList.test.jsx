import React from 'react';
import { shallow } from 'enzyme';

import UsersList from '../UsersList';
import renderer from 'react-test-renderer';

const users = [
	{
    'active': true,
    'email': 'abelthf@gmail.com',
    'id': 1,
    'username': 'abelthf'
  },
  {
    'active': true,
    'email': 'abel.huanca@upeu.edu.pe',
    'id': 2,
    'username': 'abel'
  }
];

test('UsersList renders properly', () => {
// 	shallow nos ayuda a crear el componente para luego hacer pruebas
	const wrapper = shallow(<UsersList users={ users } />);
	const element = wrapper.find('h4');
	expect(element.length).toBe(2);
	expect(element.get(0).props.children).toBe('abelthf');
});

//prueba snapshot para asegurar que la interfaz de usuario no cambia
test('UsersList renders a snapshot properly', () => {
	const tree = renderer.create(<UsersList users={ users } />).toJSON();
	expect(tree).toMatchSnapshot();
});	