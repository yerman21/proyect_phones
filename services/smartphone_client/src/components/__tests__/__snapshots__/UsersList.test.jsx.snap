// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    abelthf
  </h4>
  <h4
    className="box title is-4"
  >
    abel
  </h4>
</div>
`;
